src/Scene/BrowseTree.js

```diff
@@ -137,53 +98,38 @@ BrowseTree.prototype.browse = function(tree, camera, process, layersConfig, opti
  * @param {type} optional   : optional process
  * @returns {undefined}
  */
-BrowseTree.prototype._browse = function(node, camera, process, action, params) {
-    switch (action) {
-        case 'visibility_update':
-            {
-                if (this.quadtreeNodeVisibilityUpdate(node, camera, process, params)) {
-                    var child_action = node.isDisplayed() ? 'hide_all' : action;
-                    for (var i = 0; i < node.children.length; i++) {
-                        this._browse(node.children[i], camera, process, child_action, params);
-                    }
-                }
-            }
-            break;
-        case 'hide_all':
-            {
-                if (node.isVisible()) {
-                    this.resetQuadtreeNode(node);
-                    node.setVisibility(!process.isCulled(node, camera));
-                    node.setDisplayed(false);
-                    for (var j = 0; j < node.children.length; j++) {
-                        this._browse(node.children[j], camera, process, action, params);
-                    }
-                }
-            }
-            break;
-        case 'clean':
-            {
-                this._clean(node, node.level + 2, process, camera);
-            }
-            break;
-        default:
-            {
-                //console.error('Unknown action ', action);
-            }
+BrowseTree.prototype._browseDisplayableNode = function(node, camera, process, params) {
+    if (node.parent.isVisible() && process.processNode(node, camera, params)) {
+        if (node.isDisplayed()) {
+            this.uniformsProcess(node, camera);
+            applyFunctionToChildren(n => this._browseNonDisplayableNode(n, node.level + 2, process, camera, params), node);
+        } else {
+            applyFunctionToChildren(n => this._browseDisplayableNode(n, camera, process, params), node);
+        }
+    } else {
+        node.setVisibility(false);
+        node.setDisplayed(false);
+
+        applyFunctionToChildren(n => this._browseNonDisplayableNode(n, node.level + 2, process, camera, params), node);
     }
 };
 
-BrowseTree.prototype._clean = function(node, level, process, camera) {
+BrowseTree.prototype._browseNonDisplayableNode = function(node, level, process, camera, params) {
     // update node's sse value
     node.sse = camera.computeNodeSSE(node);
+    node.setDisplayed(false);
 
     var sse = process.checkNodeSSE(node);
 
+    if (!sse && !node.loaded) {
+        process.refineNodeLayers(node, camera, params);
```

Comment:
I should add a comment. It's meant to make sure that no tile is stuck in the `!loaded` state.